From e911023149c6d7cb03d10829af0b6467b779ec1d Mon Sep 17 00:00:00 2001
From: Andre Brait <andrebrait@gmail.com>
Date: Sat, 31 Oct 2026 04:08:30 +0000
Subject: [PATCH 2/5] luci-mod-network: add flow control advertisement option

netifd has a pause device option that advertises pause frame support and
leaves flow control to autonegotiation. The existing RX and TX options
instead force flow control with autonegotiation disabled. Add the pause
option and say how it differs from the RX and TX options.

Signed-off-by: Andre Brait <andrebrait@gmail.com>
---
 .../htdocs/luci-static/resources/tools/network.js         | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/modules/luci-mod-network/htdocs/luci-static/resources/tools/network.js b/modules/luci-mod-network/htdocs/luci-static/resources/tools/network.js
index 29edfb8..068f1b8 100644
--- a/modules/luci-mod-network/htdocs/luci-static/resources/tools/network.js
+++ b/modules/luci-mod-network/htdocs/luci-static/resources/tools/network.js
@@ -1169,6 +1169,14 @@ return baseclass.extend({
 		for (let link of devicePhySpeeds)
 			o.value('%d'.format(link), formatLinkSpeed(link));
 
+		o = this.replaceOption(s, 'devgeneral', form.ListValue, 'pause', _('Advertise flow control'),
+			_('Advertise pause frame support to the link partner. Flow control is used only if both sides support it. Setting the RX or TX option below forces flow control instead.'));
+		o.rmempty = true;
+		o.default = '';
+		o.value('', _('Default'));
+		o.value('1', _('Enable'));
+		o.value('0', _('Disable'));
+
 		o = this.replaceOption(s, 'devgeneral', form.ListValue, 'rxpause', _('Controls the receive (RX) flow control'));
 		o.rmempty = true;
 		o.default = '';
-- 
2.47.3


